### DISCUSSION TITLE
Enabling HTTP API always returns "404 page not found" on requests

### DISCUSSION THREAD

**Discussion by jamesmacwhite** [containrrr/watchtower#1192] [Q&A] [answered]
Hi,

Sorry if I'm missing something really basic, but I'm trying to enable the [HTTP API for Prometheus metric data](https://containrrr.dev/watchtower/metrics/).

I've specified the environment variables on my Docker container.

`WATCHTOWER_HTTP_API_TOKEN=xxxxxxxx`
`WATCHTOWER_HTTP_API_METRICS=true`

I've had to map the port as `8081:8080` as I have another service using port 8080.

Looking at the logs I can see the HTTP server and API has been started,  however when I browse http://[host-ip]:8081 all I get on any request to it is:

> 404 page not found

What am I doing wrong? Should I be sending any specific HTTP headers?

**Comment by jamesmacwhite** [comment, marked as answer]
Figured out the metrics endpoint is `/v1/metrics` from https://github.com/containrrr/watchtower/blob/29f5c4b254a686d9198277a78f5066666240f0e0/prometheus/prometheus.yml

Doing `curl -H "Authorization: Bearer xxxxxxxxxxx" http://host:8081/v1/metrics` worked.

I believe I've misunderstood the `/v1/update` example. This is only enabled when you set `WATCHTOWER_HTTP_API` to true?

**Reply by piksel** [reply]
Yes, this is because it was the only initial API endpoint. It's not really intuitive the way it works right now, but we don't want to break compatibility...

**Reply by jamesmacwhite** [reply]
@piksel Hi! Thanks for confirming, yes I ended up discovering this after looking through the code in the end. I did send a PR about making it clearer what the specific endpoint for metrics was in the docs, which was accepted, so hopefully it makes that part clearer. I also added a default Prometheus scrape_config example, if not using the demo.
